doc/nrf/libraries/others/event_manager_profiler.rst

```diff
@@ -0,0 +1,116 @@
+***********************************************
+
+.. note::
+	Before you complete the following steps, make sure to :ref:`implement Event Manager events and modules <event_manager_implementing_events>`.
+
+To profile an Event Manager event, you must complete the following steps:
+
+1. Enable profiling Event Manager events using the :kconfig:`CONFIG_EVENT_MANAGER_PROFILER` Kconfig option.
+#. Edit the source file for the event type:
+
+   a. Define a profiling function that logs the event data to a given buffer by calling one of the following functions for every registered data type:
+
+      * :c:func:`profiler_log_encode_uint32`
+      * :c:func:`profiler_log_encode_int32`
+      * :c:func:`profiler_log_encode_uint16`
+      * :c:func:`profiler_log_encode_int16`
+      * :c:func:`profiler_log_encode_uint8`
+      * :c:func:`profiler_log_encode_int8`
+      * :c:func:`profiler_log_encode_string`
+
+   #. Define an :c:struct:`profiler_info` structure using :c:macro:`EVENT_INFO_DEFINE` in your event source file and provide it as an argument when defining the event type with :c:macro:`EVENT_TYPE_DEFINE` macro.
+	  This structure contains a profiling function and information about the data fields that are logged.
+	  The following code example shows a profiling function for the event type ``sample_event``:
+
+	  .. code-block:: c
+
+	     static void profile_sample_event(struct log_event_buf *buf,
+		 			 const struct event_header *eh)
+		  {
+			 struct sample_event *event = cast_sample_event(eh);
+
+			 profiler_log_encode_int8(buf, event->value1);
+			 profiler_log_encode_int16(buf, event->value2);
+			 profiler_log_encode_int32(buf, event->value3);
+		   }
+
+	  The following code example shows how to define the event profiling information structure and add it to event type definition:
+
+	  .. code:: c
+
+		  EVENT_INFO_DEFINE(sample_event,
+				/* Profiled datafield types. */
+				ENCODE(PROFILER_ARG_S8, PROFILER_ARG_S16, PROFILER_ARG_S32),
+				/* Profiled data field names - displayed by profiler. */
+				ENCODE("value1", "value2", "value3"),
+				/* Function used to profile event data. */
+				profile_sample_event);
+
+		  EVENT_TYPE_DEFINE(sample_event,
+				true,
+				log_sample_event,	/* Function for logging event data. */
+				&sample_event_info);	/* Structure with data for profiling. */
+
+	  .. note::
+		  * By default, all Event Manager events that are defined with an :c:struct:`event_info` argument are profiled.
+		  * :c:struct:`sample_event_info` is defined within the :c:macro:`EVENT_INFO_DEFINE` macro.
+
+#. Use the Profiler Python scripts to profile the application.
+   See :ref:`profiler_backends` in the Profiler documentation for details.
+
+
+Event Manager profiler tracer implementation details
+****************************************************
+
+Event Manager to comunicate with Event Manager profiler tracer uses empty weak functions which are overridden in :file:`event_manager_profiler.c`.
```

Comment:
`Event Manager provides tracing hooks, that can be used to get information about Event Manager intialization, event submission and event execution in runtime.
The hooks are provided as weak functions, so that they can be overridden.
The Event Manager profiler tracer uses the hooks to register profiler events and log their occurrence when application is running.` ?

Reply:
Maybe we could explicitly list the relevant functions here (as bullet list)

Reply:
I think that it will be better to mention them in event_manager.rst and there also put list of all weak functions. In event_manager_profiler.rst I would only mention that it such weak function exist and are used by event_manager_profiler.